### PR TITLE
AG-932: Add biodomains dataset and custom transform

### PR DESCRIPTION
**Pull request by JessterB**
* Added new biodomains dataset
* Added new transform_biodomains custom transform
* Whitespace cleanup

### CODE REVIEW

**Comment by thomasyu888** [comment]
@JessterB seems like there's a file that the service account needs to agree to it's access requirements

**Comment by JessterB** [comment]
@thomasyu888  Yes, I'm aware and looking for the right path to resolve that...

File: agoradatatools/etl/transform.py [hunk 1]
```diff
@@ -352,7 +365,8 @@ def apply_custom_transformations(datasets: dict, dataset_name: str, dataset_obj:
     elif dataset_name == "distribution_data":
         return transform_distribution_data(datasets=datasets,
                                            overall_max_score=dataset_obj['custom_transformations']['overall_max_score'],
-                                           genetics_max_score=dataset_obj['custom_transformations']['genetics_max_score'],
+                                           genetics_max_score=dataset_obj['custom_transformations'][
+                                               'genetics_max_score'],
```
Comment (jaclynbeck-sage) [inline agoradatatools/etl/transform.py]: This is odd... can you put it all back on one line?

File: agoradatatools/etl/transform.py [hunk 2]
```diff
@@ -361,11 +375,15 @@ def apply_custom_transformations(datasets: dict, dataset_name: str, dataset_obj:
         return transform_rna_seq_data(datasets=datasets)
     elif dataset_name == 'gene_info':
         return transform_gene_info(datasets=datasets,
-                                   adjusted_p_value_threshold=dataset_obj['custom_transformations']['adjusted_p_value_threshold'],
-                                   protein_level_threshold=dataset_obj['custom_transformations']['protein_level_threshold'])
+                                   adjusted_p_value_threshold=dataset_obj['custom_transformations'][
+                                       'adjusted_p_value_threshold'],
+                                   protein_level_threshold=dataset_obj['custom_transformations'][
+                                       'protein_level_threshold'])
```
Comment (jaclynbeck-sage) [inline agoradatatools/etl/transform.py]: Same here, can you put these back into one line each?

File: agoradatatools/etl/transform.py [hunk 3]
```diff
@@ -341,8 +341,21 @@ def create_proteomics_distribution_data(datasets: dict) -> pd.DataFrame:
     return pd.concat(transformed)
 
 
-def apply_custom_transformations(datasets: dict, dataset_name: str, dataset_obj: dict):
+def transform_biodomains(datasets: dict):
+    biodomains = datasets['biodomains']
+    interesting_columns = ['ensembl_id', 'biodomain', 'goterms']
+    biodomains = biodomains[interesting_columns]
+
+    biodomains = biodomains.groupby(['ensembl_id', 'biodomain'])['goterms'].apply(list).reset_index()
```
Comment (jaclynbeck-sage) [inline agoradatatools/etl/transform.py]: Can you add a comment explaining what this line of code does?

File: config.yaml [hunk 4]
```diff
@@ -1,5 +1,19 @@
...
+            - name: biodomains
+              id: syn44151254.1
+              format: csv
+          final_format: json
+          custom_transformations: 1
+          provenance:
+            - syn44151254.1
+          column_rename:
+            ensg: ensembl_gene_id
+            goterm_name: goterms
```
Comment (jaclynbeck-sage) [inline config.yaml]: nitpicking a little... could this maybe be `go_terms` instead of `goterms`?

File: test_config.yaml [hunk 5]
```diff
@@ -1,5 +1,19 @@
...
+          - name: biodomains
+            id: syn44151254.1
+            format: csv
+        final_format: json
+        custom_transformations: 1
+        provenance:
+          - syn44151254.1
+        column_rename:
+          ensg: ensembl_gene_id
+          goterm_name: goterms
```
Comment (jaclynbeck-sage) [inline test_config.yaml]: Same as above.

**Review by jaclynbeck-sage — CHANGES_REQUESTED** [review]
Just a few minor edits, otherwise looks good!

File: config.yaml (hunk 4 above)
Comment (thomasyu888) [reply config.yaml]: @jaclynbeck-sage . Something we do on our team is, maybe that'll help you.

Nit: This nitpicky suggestion

File: agoradatatools/etl/transform.py (hunk 1 above)
Comment (JessterB) [reply agoradatatools/etl/transform.py]: done!

File: agoradatatools/etl/transform.py (hunk 2 above)
Comment (JessterB) [reply agoradatatools/etl/transform.py]: done!

File: agoradatatools/etl/transform.py (hunk 3 above)
Comment (JessterB) [reply agoradatatools/etl/transform.py]: done!

File: config.yaml (hunk 4 above)
Comment (JessterB) [reply config.yaml]: done!

File: test_config.yaml (hunk 5 above)
Comment (JessterB) [reply test_config.yaml]: done!

**Review by jaclynbeck-sage — APPROVED** [review]

File: agoradatatools/etl/transform.py [hunk 6]
```diff
@@ -46,17 +46,20 @@ def rename_columns(df: pd.DataFrame, column_map: dict) -> pd.DataFrame:
     return df
 
 
-def nest_fields(df: pd.DataFrame, grouping: str, new_column: str) -> pd.DataFrame:
+def nest_fields(df: pd.DataFrame, grouping: str, new_column: str, drop_columns: list = []) -> pd.DataFrame:
     """
```
Comment (jaclynbeck-sage) [inline agoradatatools/etl/transform.py:50]: Any idea why changes you merged in from dev are showing up as changes in this PR? Do I have the wrong view options selected or something?

Comment (JessterB) [reply agoradatatools/etl/transform.py:50]: No, I tried to rebase my branch on dev and it went sideways. I'm going to close this PR and open a new one shortly...

**Pull request closed without merging**